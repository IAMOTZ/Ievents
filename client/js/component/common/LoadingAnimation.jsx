import React from 'react';

// The component(animation) that is displayed for a timed action.

// This one is just an icon.
export const LoadingIcon = (props) => {
  let component;
  if (props.start) {
    component = (
      <div className="d-flex flex-column align-items-center m-3">
        <i className={`fa fa-spinner fa-pulse fa-${props.size ? props.size : 1}x`} />
      </div>
    );
  } else {
    component = null;
  }
  return component;
};

// This one is the icon inside a container.
export const LoadingContainer = props => (
  <div className="d-flex justify-content-center w-100 mt-3" style={{ minHeight: '300px' }} >
    <div style={{ marginTop: '100px' }}>
      <LoadingIcon start size={props.iconSize} />
    </div>
  </div>
);

