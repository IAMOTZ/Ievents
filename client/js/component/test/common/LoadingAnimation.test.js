import React from 'react';
import renderer from 'react-test-renderer';
import { LoadingIcon, LoadingContainer } from '../../common/LoadingAnimation.jsx';

describe('<LoadingIcon />', () => {
  it('should render nothing if start prop is not given', () => {
    const wrapper = shallow(<LoadingIcon />);
    expect(wrapper.html()).toBeNull();
  });
  it('should render a LoadingIcon of size 4', () => {
    const tree = renderer.create(<LoadingIcon start size={4} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<LoadingContainer />', () => {
  it('should render the LoadingContainer with icon size of 3', () => {
    const tree = renderer.create(<LoadingContainer iconSize={3} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
