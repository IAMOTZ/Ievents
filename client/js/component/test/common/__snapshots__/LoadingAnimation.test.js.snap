// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingContainer /> should render the LoadingContainer with icon size of 3 1`] = `
<div
  className="d-flex justify-content-center w-100 mt-3"
  style={
    Object {
      "minHeight": "300px",
    }
  }
>
  <div
    style={
      Object {
        "marginTop": "100px",
      }
    }
  >
    <div
      className="d-flex flex-column align-items-center m-3"
    >
      <i
        className="fa fa-spinner fa-pulse fa-3x"
      />
    </div>
  </div>
</div>
`;

exports[`<LoadingIcon /> should render a LoadingIcon of size 4 1`] = `
<div
  className="d-flex flex-column align-items-center m-3"
>
  <i
    className="fa fa-spinner fa-pulse fa-4x"
  />
</div>
`;
